tonic_validate/validate_scorer.py

```diff
@@ -14,50 +17,58 @@ def __init__(
+        openai_service = OpenAIService(self.model_evaluator)
+        for metric in self.metrics:
+            score = metric.score(response, openai_service)
+            scores[metric.name] = score
+
+        benchmark_item = response.benchmark_item
+        return RunData(
+            scores,
+            benchmark_item.question,
+            benchmark_item.answer,
+            response.llm_answer,
+            response.llm_context_list,
+        )
+
+    def score_run(self, responses: List[LLMResponse], parallelism=2) -> Run:
         """Calculate metric scores for a list of LLMResponse objects.
 
         Parameters
         ----------
         responses: List[LLMResponse]
             The list of LLMResponse objects to be scored.
-
+        parallelism: int
+            The number of threads to use for scoring.
+        
         Returns
         -------
         Run
             The Run object containing the scores and other data.
         """
         run_data: list[RunData] = []
-        total_scores: dict[str, float] = {}
-        score_counts: dict[str, int] = {}
-        for response in responses:
-            scores: dict[str, float] = {}
-            # We cache per response, so we need to create a new OpenAIService
-            openai_service = OpenAIService(self.model_evaluator)
-            for metric in self.metrics:
-                score = metric.score(response, openai_service)
-                scores[metric.name] = score
-                if metric.name not in total_scores:
-                    total_scores[metric.name] = 0
-                total_scores[metric.name] += score
-                if metric.name not in score_counts:
-                    score_counts[metric.name] = 0
-                score_counts[metric.name] += 1
-
-            benchmark_item = response.benchmark_item
-            run_data.append(
-                RunData(
-                    scores,
-                    benchmark_item.question,
-                    benchmark_item.answer,
-                    response.llm_answer,
-                    response.llm_context_list,
-                )
-            )
-        # Calculate overall scores
-        overall_scores: dict[str, float] = {}
-        for metric_name in total_scores:
-            overall_scores[metric_name] = (
-                total_scores[metric_name] / score_counts[metric_name]
-            )
-        return Run(overall_scores, run_data, None)
+
+        with ThreadPoolExecutor(max_workers=parallelism) as executor:
+            run_data = list(executor.map(self._score_item_rundata, responses))
+        
+
+        # Used to calculate overall score
+        total_scores: DefaultDict[str, float] = defaultdict(float)
+        num_scores: DefaultDict[str, int] = defaultdict(int)
+
+        for item in run_data:
+            for metric_name, score in item.scores.items():
+                if metric_name == AnswerSimilarityMetric.metric_name:
+                    total_scores[metric_name] += score / 5
```

Comment:
We don't need to divide by 5 here, the division by 5 will happen on the server side when calculating the overall score. I'll push a change for this.